src/nspawn/nspawn-mount.c

```diff
@@ -235,10 +253,11 @@ int bind_mount_parse(CustomMount **l, size_t *n, const char *s, bool read_only)
                         return -ENOMEM;
         }
 
-        if (isempty(source))
-                source = mfree(source);
-        else if (!source_path_is_valid(source))
-                return -EINVAL;
+        r = source_path_parse_nullable(source, &p);
+        if (r < 0)
+                return r;
+
+        free_and_replace(source, p);
```

Comment:
instead of moving p → source, why not just change the one use of `source` below to just use `p`?